laser_encoders/validate_models.py

```diff
@@ -0,0 +1,122 @@
+from laser_encoders.models import initialize_encoder
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("lang", LASER3_LANGUAGE)
+def test_validate_language_models_and_tokenize_laser3(lang):
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        print(f"Created temporary directory for {lang}", tmp_dir)
+
+        downloader = LaserModelDownloader(model_dir=tmp_dir)
+        if lang in ["kashmiri", "kas", "central kanuri", "knc"]:
+            with pytest.raises(ValueError) as excinfo:
+                downloader.download_laser3(lang)
+            assert "ValueError" in str(excinfo.value)
+            print(f"{lang} language model raised a ValueError as expected.")
+        else:
+            downloader.download_laser3(lang)
+            encoder = initialize_encoder(lang, model_dir=tmp_dir)
+            tokenizer = initialize_tokenizer(lang, model_dir=tmp_dir)
+
+            # Test tokenization with a sample sentence
+            tokenized = tokenizer.tokenize("This is a sample sentence.")
+
+    print(f"{lang} model validated successfully")
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("lang", LASER2_LANGUAGE)
+def test_validate_language_models_and_tokenize_laser2(lang):
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        print(f"Created temporary directory for {lang}", tmp_dir)
+
+        downloader = LaserModelDownloader(model_dir=tmp_dir)
+        downloader.download_laser2()
+
+        encoder = initialize_encoder(lang, model_dir=tmp_dir)
+        tokenizer = initialize_tokenizer(lang, model_dir=tmp_dir)
+
+        # Test tokenization with a sample sentence
+        tokenized = tokenizer.tokenize("This is a sample sentence.")
+
+    print(f"{lang} model validated successfully")
+
+
+class MockLaserModelDownloader:
+    def __init__(self, model_dir):
+        self.model_dir = model_dir
+
+    def get_language_code(self, language_list: dict, lang: str) -> str:
+        try:
+            lang_3_4 = language_list[lang]
+            if isinstance(lang_3_4, tuple):
+                options = ", ".join(f"'{opt}'" for opt in lang_3_4)
+                raise ValueError(
+                    f"Language '{lang_3_4}' has multiple options: {options}. Please specify using --lang."
+                )
+            return lang_3_4
+        except KeyError:
+            raise ValueError(
+                f"language name: {lang} not found in language list. Specify a supported language name"
+            )
+
+    def download_laser3(self, lang):
+        lang = self.get_language_code(LASER3_LANGUAGE, lang)
+        file_path = os.path.join(self.model_dir, f"laser3-{lang}.v1.pt")
+        if os.path.exists(file_path):
+            return False
+        else:
+            return True
+
+    def download_laser2(self):
+        files = ["laser2.pt", "laser2.spm", "laser2.cvocab"]
+        for file_name in files:
+            file_path = os.path.join(self.model_dir, file_name)
+            if os.path.exists(file_path):
+                return False
```

Comment:
This will return early won't check the other two files? (`laser2.spm` and `laser2.cvocab`)